src/Symfony/Component/Form/FormBuilderInterface.php

```diff
@@ -29,23 +29,23 @@ interface FormBuilderInterface extends \Traversable, \Countable, FormConfigBuild
      *
      * @return self
      */
-    public function add($child, $type = null, array $options = []);
+    public function add($child, string $type = null, array $options = []);
 
     /**
      * Creates a form builder.
      *
-     * @param string      $name    The name of the form or the name of the property
+     * @param string|int  $name    The name of the form or the name of the property
```

Comment:
int feels a lot strange here:

look at this code:

1. this is FormBuilder implementation of create 
```php
    public function create($name, $type = null, array $options = [])
    {
        if ($this->locked) {
            throw new BadMethodCallException('FormBuilder methods cannot be accessed anymore once the builder is turned into a FormConfigInterface instance.');
        }

        if (null === $type && null === $this->getDataClass()) {
            $type = 'Symfony\Component\Form\Extension\Core\Type\TextType';
        }

        if (null !== $type) {
            return $this->getFormFactory()->createNamedBuilder($name, $type, null, $options);
        }

        return $this->getFormFactory()->createBuilderForProperty($this->getDataClass(), $name, null, $options);
    }
```

2. and this is `FormFactoryInterface::createBuilerForProperty`
```php
    /**
     * Returns a form builder for a property of a class.
     *
     * If any of the 'required' and type options can be guessed,
     * and are not provided in the options argument, the guessed value is used.
     *
     * @param string $class    The fully qualified class name
     * @param string $property The name of the property to guess for
     * @param mixed  $data     The initial data
     * @param array  $options  The options for the builder
     *
     * @return FormBuilderInterface The form builder named after the property
     *
     * @throws \Symfony\Component\OptionsResolver\Exception\InvalidOptionsException if any given option is not applicable to the form type
     */
    public function createBuilderForProperty($class, $property, $data = null, array $options = []);
```

3. and this is it's implementation in `FormBuilder`
```php
    public function createBuilderForProperty($class, $property, $data = null, array $options = [])
    {
        if (null === $guesser = $this->registry->getTypeGuesser()) {
            return $this->createNamedBuilder($property, 'Symfony\Component\Form\Extension\Core\Type\TextType', $data, $options);
        }

        $typeGuess = $guesser->guessType($class, $property);
        $maxLengthGuess = $guesser->guessMaxLength($class, $property);
        $requiredGuess = $guesser->guessRequired($class, $property);
        $patternGuess = $guesser->guessPattern($class, $property);

        $type = $typeGuess ? $typeGuess->getType() : 'Symfony\Component\Form\Extension\Core\Type\TextType';

        $maxLength = $maxLengthGuess ? $maxLengthGuess->getValue() : null;
        $pattern = $patternGuess ? $patternGuess->getValue() : null;

        if (null !== $pattern) {
            $options = array_replace_recursive(['attr' => ['pattern' => $pattern]], $options);
        }

        if (null !== $maxLength) {
            $options = array_replace_recursive(['attr' => ['maxlength' => $maxLength]], $options);
        }

        if ($requiredGuess) {
            $options = array_merge(['required' => $requiredGuess->getValue()], $options);
        }

        // user options may override guessed options
        if ($typeGuess) {
            $attrs = [];
            $typeGuessOptions = $typeGuess->getOptions();
            if (isset($typeGuessOptions['attr']) && isset($options['attr'])) {
                $attrs = ['attr' => array_merge($typeGuessOptions['attr'], $options['attr'])];
            }

            $options = array_merge($typeGuessOptions, $options, $attrs);
        }

        return $this->createNamedBuilder($property, $type, $data, $options);
    }
```

4. note that it uses `FormTypeGuesserInterface::guessType`
```php
    /**
     * Returns a field guess for a property name of a class.
     *
     * @param string $class    The fully qualified class name
     * @param string $property The name of the property to guess for
     *
     * @return Guess\TypeGuess|null A guess for the field's type and options
     */
    public function guessType($class, $property);
```

this leads to that int can be a property name :|

Reply:
We already discussed the issue and we'll do strings only version, see https://github.com/symfony/symfony/pull/32400. 

Nevertheless, using `int` in both Forms and Builders is possible because the name later is always converted to `string` in https://github.com/symfony/symfony/blob/4.3/src/Symfony/Component/Form/FormConfigBuilder.php#L126. Btw, internally Symfony does pass ints to all these methods, see tests covering these cases. You can also try `$builder->add(10)` in your project — the resulting form will have `'10'`, so it works. That is why my idea is not nonsense.

But I agree that if we design the component from scratch, we would allow only strings. So that's what we want to achieve in 5.0.